src/helpers.tsx

```diff
@@ -1,29 +1,50 @@
 /* eslint-disable import/prefer-default-export */
+import { logger } from '@storybook/client-logger';
 import { WrapperSettings } from '@storybook/addons';
 import { DEFAULT_PADDING } from './constants';
 
-type Option = {
-  name: string;
-  value: string;
-  default?: boolean;
+type Value = { name: string; value: string };
+
+type Values = { [key: string]: Value };
+
+type PaddingConfig = {
+  default: string;
+  disable?: boolean;
+  values: Values;
 }
 
-type Options = Option[] | WrapperSettings['parameters'];
+type Option = Value & {default?: boolean}
+
+type Options = Option[] | PaddingConfig | WrapperSettings['parameters'];
+
+export const normalizeValues = (options: Options) => (Array.isArray(options)
+  ? options
+  : Object.entries<Value>(options.values).map(([key, { name, value }]) => {
+    const isDefault = options.default === key;
 
-export const isEnabled = (options: Options): options is Option[] => options.length > 0;
+    return { name, value, default: isDefault };
+  }));
 
-export const getSelectedPadding = (options: Options, currentValue: string): string => {
-  if (!isEnabled(options)) {
-    return DEFAULT_PADDING;
+export const isEnabled = (options: Options) => {
+  if (Array.isArray(options)) {
+    logger.warn(
+      'Storybook addon paddings has changed, please migrate to the new recommended configuration.',
+    );
+
+    return options.length > 0;
   }
 
+  return options && !options?.disable && Object.entries(options.values).length > 0;
+};
+
+export const getSelectedPadding = (values: Option[], currentValue: string): string => {
```

Comment:
I think the logic inside `getSelectedPadding` was getting a bit complex given the access of other helper functions.
What I did is:
1. Ensure it always received the same collection of elements it needs, to pick a padding.
2. The consumers `index` and `register`, will ensure to validate`(isEnable)` and pass the right data `(normalizeValues)`.

This avoids the need of calling helpers internally and in the consumers as well.

Reply:
Sounds good 👍